Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -631,139 +625,44 @@ public void OnFocusExit(FocusEventData eventData)
 
         #endregion MixedRealityFocusHandlers
 
-        #region MixedRealityPointerHandlers
-
         /// <summary>
-        /// pointer up event has fired
+        /// Starts a timer to check if input is in progress
+        ///  - Make sure global pointer events are not double firing
+        ///  - Make sure Global Input events are not double firing
+        ///  - Make sure pointer events are not duplicating an input event
         /// </summary>
-        /// <param name="eventData"></param>
-        public void OnPointerUp(MixedRealityPointerEventData eventData)
+        /// <param name="isFromInputDown"></param>
+        protected void StartClickTimer(bool isFromInputDown = false)
         {
-            pointerInputAction = eventData.MixedRealityInputAction;
-            if ((!CanInteract() && !HasPress))
+            if (IsGlobal || isFromInputDown)
             {
-                return;
-            }
-
-            bool isGrab = false;
-            if (ShouldListen(eventData, out isGrab))
-            {
-                SetPress(false);
-                if (isGrab)
+                if (clickValidTimer != null)
                 {
-                    // what if we have two hands grabbing?
-                    SetGrab(false);
+                    StopCoroutine(clickValidTimer);
+                    clickValidTimer = null;
                 }
 
-                SetGesture(false);
-
-                eventData.Use();
+                clickValidTimer = StartCoroutine(InputDownTimer(clickTime));
             }
         }
 
-        /// <summary>
-        /// Pointer down event has fired
-        /// </summary>
-        /// <param name="eventData"></param>
-        public void OnPointerDown(MixedRealityPointerEventData eventData)
-        {
-            pointerInputAction = eventData.MixedRealityInputAction;
-            if (!CanInteract())
-            {
-                return;
-            }
-
-            bool isGrab = false;
-            if (ShouldListen(eventData, out isGrab))
-            {
-                SetPress(true);
-                SetGrab(isGrab);
-
-                dragStart = eventData.Pointer.Position;
-                eventData.Use();
-            }
-        }
-
-        public void OnPointerDragged(MixedRealityPointerEventData eventData)
+        protected void StopClickTimer()
         {
-            bool isGrab = false;
-            if (!HasGesture && CanInteract() && (ShouldListen(eventData, out isGrab) || eventData.MixedRealityInputAction.Description == "None") && (dragStart - eventData.Pointer.Position).magnitude > 0.1f)
-            {
-                SetGesture(true);
```

Comment:
After this PR interactable would no longer set "is gesture" to true. Need to add that back in.